rabbitmq/src/test/groovy/io/micronaut/rabbitmq/connect/TemporarilyDownConsumersSpec.groovy

```diff
@@ -0,0 +1,321 @@
+package io.micronaut.rabbitmq.connect
+
+import com.rabbitmq.client.BlockedListener
+import com.rabbitmq.client.ShutdownListener
+import io.micronaut.context.ApplicationContext
+import io.micronaut.context.annotation.Requires
+import io.micronaut.core.io.socket.SocketUtils
+import io.micronaut.rabbitmq.annotation.Binding
+import io.micronaut.rabbitmq.annotation.Queue
+import io.micronaut.rabbitmq.annotation.RabbitClient
+import io.micronaut.rabbitmq.annotation.RabbitListener
+import io.micronaut.rabbitmq.connect.recovery.TemporarilyDownException
+import io.micronaut.rabbitmq.connect.recovery.TemporarilyDownIOException
+import io.micronaut.rabbitmq.connect.recovery.TemporarilyDownRuntimeException
+import io.micronaut.rabbitmq.exception.RabbitClientException
+import io.micronaut.rabbitmq.exception.RabbitListenerException
+import io.micronaut.rabbitmq.exception.RabbitListenerExceptionHandler
+import org.testcontainers.containers.FixedHostPortGenericContainer
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy
+import spock.lang.Specification
+import spock.util.concurrent.PollingConditions
+
+class TemporarilyDownConsumersSpec extends Specification {
+
+    static PollingConditions conditions = new PollingConditions(timeout: 10)
+
+    void "test temporarily down consumer"() {
+        int port = SocketUtils.findAvailableTcpPort()
+        FixedHostPortGenericContainer rabbitmq = new FixedHostPortGenericContainer("library/rabbitmq:3.7")
```

Comment:
Do we support test containers?

Reply:
> Do we support test containers?

@dstepanov If you have a better way to test this scenario, I'm all ears:

1. First, the application context is started _while the server is down_.
2. Then, RabbitMQ beans try to connect to the server _on port X_.
3. Finally, the server starts listening _on port X_.